apps/condo/domains/meter/schema/MeterReading.js

```diff
@@ -0,0 +1,193 @@
+            kmigratorOptions: { null: true, on_delete: 'models.SET_NULL' },
+        },
+
+        clientName: {
+            schemaDoc: 'Inhabitant/customer/person who has a problem. Sometimes we get a problem from an unregistered client, in such cases we have a null inside the `client` and just have something here. Or sometimes clients want to change it',
+            type: Text,
+        },
+
+        clientEmail: {
+            schemaDoc: 'Inhabitant/customer/person who has a problem. Sometimes we get a problem from an unregistered client, in such cases we have a null inside the `client` and just have something here. Or sometimes clients want to change it',
+            type: Text,
+        },
+
+        clientPhone: {
+            schemaDoc: 'Inhabitant/customer/person who has a problem. Sometimes we get a problem from an unregistered client, in such cases we have a null inside the `client` and just have something here. Or sometimes clients want to change it',
+            type: Text,
+            hooks: {
+                resolveInput: async ({ resolvedData }) => {
+                    if (!resolvedData['clientPhone']) return resolvedData['clientPhone']
+                    const newValue = normalizePhone(resolvedData['clientPhone'])
+                    return newValue || resolvedData['clientPhone']
+                },
+                validateInput: async ({ resolvedData, addFieldValidationError }) => {
+                    const newValue = normalizePhone(resolvedData['clientPhone'])
+                    if (resolvedData['clientPhone'] && newValue !== resolvedData['clientPhone']) {
+                        addFieldValidationError(`${PHONE_WRONG_FORMAT_ERROR}phone] invalid format`)
+                    }
+                },
+            },
+        },
+
+        source: {
+            schemaDoc: 'Meter reading source channel/system. Examples: call, mobile_app, billing, ...',
+            type: Relationship,
+            ref: 'MeterReadingSource',
+            isRequired: true,
+            knexOptions: { isNotNullable: true }, // Required relationship only!
+            kmigratorOptions: { null: false, on_delete: 'models.PROTECT' },
+        },
+
+    },
+
+    hooks: {
+        validateInput: async ({ context, operation, itemId, addFieldValidationError }) => {
+            // if 2 minutes have passed since the moment the readings were taken, then they cannot be changed
+            if (operation === 'update') {
+                const [meterReading] = await MeterReadingApi.getAll(context, { id: itemId }, {
+                    sortBy: ['createdAt_DESC'],
+                })
+                const createdDate = moment(meterReading.createdAt)
+                const now = moment()
+                const diffInMinutes = now.diff(createdDate, 'minutes')
+
+                if (!meterReading || diffInMinutes > 2) {
+                    addFieldValidationError(`${TIME_IS_UP_ERROR}date] 2 minutes have passed since MeterReading was created`)
```

Comment:
```suggestion
                    addFieldValidationError(`${UPDATE_TIMEOUT_EXPIRED_ERROR}date] 2 minutes have passed since MeterReading was created`)
```

Reply:
This logic has been removed at the moment, we decided that `MeterReadings` cannot be edited at all